### PR TITLE
Executable CLI + Refreshed Docs

### PR DESCRIPTION
**Pull request by jrasm91**
Executable CLI: add the `bin` section to `package.json`. Now `js/src/shc-validator` is published to node's `bin` folder after package installation, making it available system-wide (assuming the bin folder is already in the `PATH`).

Update the docs:
- make npm install instructions more obvious
- make examples/usage more obvious
- update usage to use `shc-validator` executable
- add markdown tables for command options
- decrease (unnecessary) code indentation

Example `shc-validator` command:
image

### CODE REVIEW

File: src/shc-validator.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/env node
```
Comment (jmandel) [inline src/shc-validator.ts:1]: Neat, I didn't know about https://github.com/microsoft/TypeScript/issues/2749

File: README.md [hunk 2]
```diff
@@ -4,130 +4,267 @@
...
+npm install github:smart-on-fhir/health-cards-validation-SDK#v1.0.0-4
+```
 
-The tests download and validate, among other things, the spec [examples](https://smarthealth.cards/examples/). A breaking spec change might invalidate the downloaded examples, which can be refreshed using:
+## Examples
 
-                npm run fetch-examples -- --force
+### Validate a `.smart-health-card` file (`data.smart-health-card`)
+```
+smc-validator --path data.smart-health-card --type healthcard
```
Comment (christianpaquin) [inline README.md:41]: shc-validator (instead of smc-validator); here and elsewhere in the page.

Comment (jrasm91) [reply README.md]: Good catch, I've updated it to `shc` everywhere.

**Comment by ljoy913** [comment]
@jrasm91 
Hi. What version of npm are you using?  When I use npm 7.x and do a global install, I get a bunch of errors (npm 6.x or a local install works fine):
`npm i -g github:Pravici/health-cards-validation-SDK`

I don't think it has anything to do with your PR.  I'm just trying to determine if it's an npm issue, a me issue, or a project config issue.

There are a bunch of TypeScript errors from not being able to find modules when building during the __prepare__ script.

I found similar issues online with __prepare__,/npm 7 when performing a github install, but nothing about it being specific to global installations.

**Comment by jrasm91** [comment]
```
npm -v
6.13.4
```
```
node -v
v12.14.1
```

**Comment by jrasm91** [comment]
@ljoy913 - I upgraded to node 16 / npm 7 and I now get this following error, so I don't think it's specific to your setup.
```
npm ERR! npm ERR! sh: 1: tsc: not found
```

In general, it looks like code is being built locally on the user's machine (at install time), oppose to at publish time. I think people generally do any pre-building tasks prior to publishing the package to avoid these types of issues.

For example, `qrcode` is build via rollup prior to publishing to npm.
https://github.com/soldair/node-qrcode/blob/master/package.json#L34

**Comment by christianpaquin** [comment]
Sorry for the delay here; we had some urgent work last week. We'll split this PR in two, one to address the bin install, and one to reorg the README.

**Comment by jrasm91** [comment]
It's quite frustrating that I took the time to open a pull request and suggest improvements, only to be ghosted for two weeks, and then have my changes copied into another PR with no attribution (https://github.com/smart-on-fhir/health-cards-dev-tools/pull/153).

**Comment by ljoy913** [comment]
My apologies.  We were going to break your PR up into a separate 'bin' pr and a 'readme' pr to keep the updates functionally atomic.  
We were still in the process of getting the Readme updated as we need to add additional notes regarding an npm bug preventing git-repo global install on Node 15+ and exploring potential workarounds.
As we couldn't directly update your PR, we needed to create a new readme pr (yet to be done) to add these additional notes once I got a workaround resolved.

Sorry for not discussing these decisions with you prior to updating the project.

**Comment by jmandel** [comment]
Indeed, your contributions here are most welcome and I hope this won't put you off of future contributions -- please chalk this up to the fact that we have not had many external contributors yet and haven't ironed out a smooth process for reviews.

We can and will do better! In the meantime this sounds like a great opportunity for us to add an explicit contributors list to our readme. Are there other steps that you think would be helpful, without rewriting history on our main branch @jrasm91 ?

**Comment by jrasm91** [comment]
@ljoy913 - Atomic commits are fine. I even offered to split up the PR [here](https://github.com/smart-on-fhir/health-cards-dev-tools/issues/140#issuecomment-902178722), in the initial issue. Also, if you had asked, I could have given you access to the branch/PR.

@jmandel  - I kind of assumed that contributions were welcome via the standard github workflow: fork the repo, do changes, open a pull request, use comments, etc. to request changes, and eventually merge in the PR (which would auto-populate the contributors section of the repo). Maybe consider adding a [CONTRIBUTING.md](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors#adding-a-contributing-file) if you don't plan on following that flow.

> ... We'll split this PR in two, one to address the bin install, and one to reorg the README.

It looks like you will take care of implementing the rest of the changes, so I've closed the PR.

**Pull request closed without merging**